### ISSUE TITLE
Provide a way to set a default selection

### ISSUE THREAD

**Issue by cpsievert** [rstudio/crosstalk#16]
I'm currently doing this via the `defaultValues` argument in plotly's `highlight()` function, and have found it useful for sharing interesting selection states, but this should really be a property of the data, not the plot.

**Comment by cpsievert** [comment]
In fact, it'd be neat if the `selection()` method could be used for this...

**Comment by jcheng5** [comment]
The problem is that in a non-Shiny world, there's not really a "data" entity in the page at all, is there? There's only the projections of the data onto the plot.

Maybe if you explicitly add a directive to the page that says "Make a selection using these keys and this group".

```r
sd <- SharedData(mtcars)

d3scatter(sd, ...)
d3scatter(sd, ...)
setDefaultSelection(sd$groupName(), row.names(mtcars %>% filter(mpg > 30)))
```

It's not elegant (imperative instead of declarative, standalone instead of part of the SharedData class) but I don't know where else you would hang this logic?

**Comment by przemo** [comment]
I guess it would be great to include it in filter functions, e.g. filter_select(default='...')

**Comment by kent37** [comment]
Yes please. Setting the initial state is useful for creating a default. For example a map might show many years of data with the current year being the most generally useful and interesting. Initializing checkboxes to show just one year would be helpful.

**Comment by cpsievert** [comment]
> in a non-Shiny world, there's not really a "data" entity in the page at all, is there? 

Do we really need one? We have sufficient information to set the relevant value(s) at print time, so why not do something [like this](https://gist.github.com/cpsievert/5b0bb95260e4fd890de37ae620e11c4a)? 

Note that plotly responds to the callback, but `filter_select()` does not. Ideally, `filter_select()` and friends should be able to *respond* to events as well as set them, but that is kind of a separate issue, so opening #34.

**Comment by jcheng5** [comment]
@cpsievert The `set_values` is along the lines of what I meant by adding a directive to the page.

**Comment by ewallace** [comment]
Yes, we really need one.

The problem I'm having involves making an interactive plot for 5000+ items in time series, so I can compare 2-10 time series in different colours. The existing default is to plot all of them to start with, and that plot generation crashes my R session in knitR. It would be great to be able to select default values to plot to avoid that. 

(Unfortunately I did not understand your set-values.R code).

**Comment by bwlewis** [comment]
I agree that sometimes initialization is super useful.  I think that it's possible to initialize widget selections (and filters) without needing to modify the currently lovely minimalist crosstalk interface.

See for example the 2nd example here: https://bwlewis.github.io/crosstool/

That package is still a bit sketchy but manages to initialize a selection without modifying the current crosstalk api.

It can do other wonky things like transcode keys between groups, etc. It doesn't support filter selections yet, but I think that can be added in an obvious way.

Anyway, just one possible idea for this.

**Comment by bwlewis** [comment]
Let me add a bit of explanation about my thinking. Although I understand that crosstalk is intentionally opinionated with a bias for selecting and filtering, I plan to use (abuse?) it for other purposes too. I tend to see crosstalk more generically as a communication system with two channels, filter handle and select handle.

**Comment by jcheng5** [comment]
I would love a PR for default selection on the filter inputs. Defaults for filters don't have the same difficulties as defaults for selections.

**Comment by bwlewis** [comment]
yeah, still thinking and experimenting...

**Comment by EnricowithR** [comment]
I also think a default selection would be really useful, are there any solutions for that so far?

**Comment by joelgombin** [comment]
What's the current state of affairs regarding this issue?

**Comment by abiyug** [comment]
Yes, what is the latest? Can we set "initial state" for filter_select from crostalk?

**Comment by jcheng5** [comment]
@abiyug still in pr form https://github.com/rstudio/crosstalk/pull/70

**Comment by abiyug** [comment]
Just tested "selected" in flex_dash frame work with devtools version crosstalk_1.0.1.9000. not working here.  Any ETA?

**Comment by jcheng5** [comment]
You tried installing `”rstudio/crosstalk#70”` and it didn’t work? If so, details or a repro case on that PR would be great, thanks.

**Comment by abiyug** [comment]
#70 works!

**Comment by abiyug** [comment]
It "selected" option works with leaflet and DT, but not with plotly any guidance @cpsievert ?

**Comment by cpsievert** [comment]
You're right, it doesn't appear to work with plotly, here's a minimal example:

```r
library(crosstalk)
library(plotly)

sd <- SharedData$new(mtcars)
filt <- filter_select(
  id = "mag",
  label = "Select cylinder",
  sharedData = sd,
  group = ~cyl,
  multiple = FALSE,
  selected = 6
)

p <- plot_ly(sd, x = ~wt, y = ~mpg, color = ~as.factor(cyl)) %>% 
  add_markers()

bscols(widths = c(3,9), filt, p)
```

**Comment by cthomson8** [comment]
Has there been any progress on making this work with plotly?

**Comment by stvrd** [comment]
Kudos for your great work! I agree that the selected argument would be a major improvement, hope to see it soon working with plotly.

**Comment by abiyug** [comment]
This is such an important feature. Any update?

**Comment by logan4data** [comment]
Any updates, guidance on this pls,

**Comment by dexterpante** [comment]
Hello, any updates on this? This is a good improvement if selected option is added.
